Build failing on Ubuntu 18.04.1 via WLS

I'm using WSL on Windows 10 with Ubuntu 18.04. Getting an odd compilation error. OpenGL is installed correctly and I am able to run OpenGL apps over X11 remoting via VcXsrv.

-------------------------------------------------------------------------------------------------------------------

/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: The "Nerdbank.GitVersioning.Tasks.GetBuildVersion" task failed unexpectedly. [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: Unable to load shared library '/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/MSBuildFull/lib/linux/x86_64/libgit2-1196807.so' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: /usr/lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/MSBuildFull/lib/linux/x86_64/libgit2-1196807.so) [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.InternalLoadUnmanagedDllFromPath(String unmanagedDllPath) [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.LoadUnmanagedDllFromPath(String unmanagedDllPath) [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at MSBuildExtensionTask.ContextAwareTask.CustomAssemblyLoader.LoadUnmanagedDll(String unmanagedDllName) [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.ResolveUnmanagedDll(String unmanagedDllName, IntPtr gchManagedAssemblyLoadContext) [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.LibraryLifetimeObject..ctor() [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods..cctor() [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    --- End of inner exception stack trace --- [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_opts(Int32 option, UInt32 level, String path) [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.GlobalSettings.SetConfigSearchPaths(ConfigurationLevel level, String[] paths) [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.OpenGitRepo(String repoRoot) [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.Create(String projectDirectory, String gitRepoDirectory, ICloudBuild cloudBuild, Nullable`1 overrideBuildNumberOffset, String projectPathRelativeToGitRepoRoot) [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.Tasks.GetBuildVersion.ExecuteInner() [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: The "Nerdbank.GitVersioning.Tasks.GetBuildVersion" task failed unexpectedly. [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: Unable to load shared library '/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/MSBuildFull/lib/linux/x86_64/libgit2-1196807.so' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: /usr/lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/MSBuildFull/lib/linux/x86_64/libgit2-1196807.so) [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.InternalLoadUnmanagedDllFromPath(String unmanagedDllPath) [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.LoadUnmanagedDllFromPath(String unmanagedDllPath) [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at MSBuildExtensionTask.ContextAwareTask.CustomAssemblyLoader.LoadUnmanagedDll(String unmanagedDllName) [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.ResolveUnmanagedDll(String unmanagedDllName, IntPtr gchManagedAssemblyLoadContext) [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.LibraryLifetimeObject..ctor() [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods..cctor() [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    --- End of inner exception stack trace --- [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_opts(Int32 option, UInt32 level, String path) [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.GlobalSettings.SetConfigSearchPaths(ConfigurationLevel level, String[] paths) [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.OpenGitRepo(String repoRoot) [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.Create(String projectDirectory, String gitRepoDirectory, ICloudBuild cloudBuild, Nullable`1 overrideBuildNumberOffset, String projectPathRelativeToGitRepoRoot) [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.Tasks.GetBuildVersion.ExecuteInner() [/home/clandress/veldrid/src/Veldrid.SDL2/Veldrid.SDL2.csproj]
MSBUILD : error MSB4166: Child node "6" exited prematurely. Shutting down. Diagnostic information may be found in files in "/tmp/" and will be named MSBuild_*.failure.txt. This location can be changed by setting the MSBUILDDEBUGPATH environment variable to a different directory.
MSBUILD : error MSB4166: Child node "2" exited prematurely. Shutting down. Diagnostic information may be found in files in "/tmp/" and will be named MSBuild_*.failure.txt. This location can be changed by setting the MSBUILDDEBUGPATH environment variable to a different directory.
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: The "Nerdbank.GitVersioning.Tasks.GetBuildVersion" task failed unexpectedly. [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: Unable to load shared library '/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/MSBuildFull/lib/linux/x86_64/libgit2-1196807.so' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: /usr/lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/MSBuildFull/lib/linux/x86_64/libgit2-1196807.so) [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.InternalLoadUnmanagedDllFromPath(String unmanagedDllPath) [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.LoadUnmanagedDllFromPath(String unmanagedDllPath) [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at MSBuildExtensionTask.ContextAwareTask.CustomAssemblyLoader.LoadUnmanagedDll(String unmanagedDllName) [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.ResolveUnmanagedDll(String unmanagedDllName, IntPtr gchManagedAssemblyLoadContext) [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.LibraryLifetimeObject..ctor() [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods..cctor() [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    --- End of inner exception stack trace --- [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_opts(Int32 option, UInt32 level, String path) [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.GlobalSettings.SetConfigSearchPaths(ConfigurationLevel level, String[] paths) [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.OpenGitRepo(String repoRoot) [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.Create(String projectDirectory, String gitRepoDirectory, ICloudBuild cloudBuild, Nullable`1 overrideBuildNumberOffset, String projectPathRelativeToGitRepoRoot) [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.Tasks.GetBuildVersion.ExecuteInner() [/home/clandress/veldrid/src/Veldrid.MetalBindings/Veldrid.MetalBindings.csproj]

Build FAILED.

/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: The "Nerdbank.GitVersioning.Tasks.GetBuildVersion" task failed unexpectedly. [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]

/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoUnhandled Exception:undException: Unable to load shared library '/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/MSBuildFull/lib/linux/x86_64/libgit2-1196807.so' or one of its dependencies. In order to help diagnose loading problems, consider setting the L D_DEBUG environment variable: /usr/lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/MSBuildFull/lib/linux/x86_64/libgit2-1196807.so) [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: Unable to load shared library '/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/MSBuildFull/lib/linux/x86_64/libgit2-1196807.so' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: /usr/lib/x86_64-linux-gnu/libcurl.so.4: version `CURL_OPENSSL_3' not found (required by /home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/MSBuildFull/lib/linux/x86_64/libgit2-1196807.so)
   at System.Runtime.Loader.AssemblyLoadContext.InternalLoadUnmanagedDllFromPath(String unmanagedDllPath)
   at System.Runtime.Loader.AssemblyLoadContext.LoadUnmanagedDllFromPath(String unmanagedDllPath)
   at MSBuildExtensionTask.ContextAwareTask.CustomAssemblyLoader.LoadUnmanagedDll(String unmanagedDllName)
   at System.Runtime.Loader.AssemblyLoadContext.ResolveUnmanagedDll(String unmanagedDllName, IntPtr gchManagedAssemblyLoadContext)
   at LibGit2Sharp.Core.NativeMethods.git_libgit2_init()
   at LibGit2Sharp.Core.NativeMethods.LibraryLifetimeObject..ctor()
   at LibGit2Sharp.Core.NativeMethods..cctor()
   --- End of inner exception stack trace ---
   at LibGit2Sharp.Core.NativeMethods.RemoveHandle()
   at LibGit2Sharp.Core.NativeMethods.LibraryLifetimeObject.Finalize()/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.InternalLoadUnmanagedDllFromPath(String unmanagedDllPath) [/home/clandress/veldrid/src/
Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.LoadUnmanagedDllFromPath(String unmanagedDllPath) [/home/clandress/veldrid/src/Veldrid.OpenGLBindings/Veldrid.OpenGLBindings.csproj]
/home/clandress/.nuget/packages/nerdbank.gitversioning/2.1.23/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at MSBuildExtensionTask.ContextAwareTask.CustomAssemblyLoader.LoadUnmanagedDll(String unmanagedDllName) [/home/clandress/veldrid/src

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -37,7 +37,7 @@
   </ItemGroup>
 
   <PropertyGroup>
-    <NerdbankGitVersioningVersion>2.2.33</NerdbankGitVersioningVersion>
+    <NerdbankGitVersioningVersion>2.3.38</NerdbankGitVersioningVersion>
   </PropertyGroup>
 
   <ItemGroup>

